### DISCUSSION TITLE
Decouple buttons from relays only when MQTT is connected

### DISCUSSION THREAD

**Discussion by avishorp** [arendst/Tasmota#15718] [Q&A] [answered]
I use SetOption73 to decouple between buttons and relays (SONOFF Touch). The problem is that if the MQTT is not connected, there is no way to control the relays. Is it possible to break the button-relay coupling only when MQTT is connected?

**Comment by barbudor** [comment, marked as answer]
You could also use rules to manage the coupling
But your system would become a bit complicated if switching is performed from MQTT in one case and from local in other case
If you need it to work when MQTT is not connected, why decoupling in 1st place ?

**Comment by avishorp** [comment]
The system works primarily with MQTT connected, but I don't want to get
stuck in case of a malfunction, in such a way that I cannot turn on or off
the lights. I would prefer a "fallback" mode that gives some kind of
limited access.

‫בתאריך יום ב׳, 30 במאי 2022 ב-15:55 מאת ‪Barbudor‬‏ <‪
***@***.***‬‏>:‬

**Reply by barbudor** [reply]
So you may look into ButtonTopic instead of SetOption73 as it allows to define a custom topic for Button which falls-back to original behavior in case of MQTT disconnection

The info about that is documented only in SwitchTopic but applies to both:
https://tasmota.github.io/docs/Buttons-and-Switches/#switchtopic

**Reply by timaseth** [reply]
Another option (not a great one) is to set SO73 to 0 when mqtt disconnects and viceversa!
A simple rule would do that.
